flink-table/flink-table-planner/src/test/java/org/apache/flink/table/planner/plan/nodes/exec/testutils/RestoreTestBase.java

```diff
@@ -144,7 +150,42 @@ private Stream<Arguments> createSpecs() {
         return getAllMetadata().stream()
                 .flatMap(
                         metadata ->
-                                supportedPrograms().stream().map(p -> Arguments.of(p, metadata)));
+                                supportedPrograms().stream()
+                                        .flatMap(
+                                                program ->
+                                                        getSavepointPaths(program, metadata)
+                                                                .map(
+                                                                        savepointPath ->
+                                                                                Arguments.of(
+                                                                                        program,
+                                                                                        getPlanPath(
+                                                                                                program,
+                                                                                                metadata),
+                                                                                        savepointPath))));
+    }
+
+    /**
+     * The method can be overridden in a subclass to test multiple savepoint files for a given
+     * program and a node in a particular version. This can be useful e.g. to test a node against
+     * savepoint generated in different Flink versions.
+     */
+    protected Stream<String> getSavepointPaths(
+            TableTestProgram program, ExecNodeMetadata metadata) {
+        return Stream.of(getSavepointPath(program, metadata, null).toString());
+    }
+
+    protected final Path getSavepointPath(
```

Comment:
nit: return `String` to match with `getSavepointPaths`. also avoid the additional toString in the `OverWindowRestoreTest `

Reply:
the `Paths.get` can be done in `generateTestSetupFiles `